src/Validator/IsInt.php

```diff
@@ -103,6 +144,9 @@ public function isValid($value)
 
         if (is_int($value)) {
             return true;
+        } elseif (self::COMPARE_STRICT == $this->strict) {
+            $this->error(self::INVALID);
```

Comment:
I wonder if we should introduce a new error type for this to ensure distinction (e.g. `self::NOT_STRICTLY_INT` for example)